Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -61,32 +50,38 @@ class SpriteRepository {
         }
     }
     
-    func updateSprite(styleURI: StyleURI, baseURL: URL, completion: @escaping CompletionHandler) {
+    func updateInstruction(for instruction: VisualInstructionBanner, completion: @escaping CompletionHandler) {
+        guard getSpriteImage() == nil else {
+            completion()
+            return
+        }
+        updateSprite(styleURI: self.styleURI, completion: completion)
+    }
+    
+    func updateSprite(styleURI: StyleURI, completion: @escaping CompletionHandler) {
         spriteCache.clearMemory()
         infoCache.clearMemory()
         
-        // Update the styleURI and baseURL just after the Sprite memory reset. When the connection is poor, the next round of style update
+        // Update the styleURI just after the Sprite memory reset. When the connection is poor, the next round of style update
         // or the representation update could use the correct ones.
         self.styleURI = styleURI
-        self.baseURL = baseURL
         
         guard let styleID = styleURI.rawValue.components(separatedBy: "styles")[safe: 1],
-              let infoRequestURL = spriteURL(isImage: false, baseURL: baseURL, styleID: styleID),
-              let spriteRequestURL = spriteURL(isImage: true, baseURL: baseURL, styleID: styleID) else {
+              let infoRequestURL = spriteURL(isImage: false, styleID: styleID),
+              let spriteRequestURL = spriteURL(isImage: true, styleID: styleID) else {
                   completion()
                   return
               }
         
-        let spriteKey = "\(styleID)-\(baseURL.absoluteString)"
         let dispatchGroup = DispatchGroup()
         
         dispatchGroup.enter()
-        downloadInfo(infoRequestURL, spriteKey: spriteKey) { (_) in
+        downloadInfo(infoRequestURL, spriteKey: styleID) { (_) in
```

Comment:
Because the `baseURL` as a constant, only using `styleID` as the cache key for Sprite.